### PR TITLE
op-chain-ops: update config

### PR DESCRIPTION
**Pull request by tynes**
**Description**
We need config options for the recipients of the fee vaults. This adds it to the config parsing.

Also adds a helper function to read the correct deployment
artifacts from disk given a `hardhat.Hardhat` instance.
This code was being repeated throughout the codebase
so its better to consolidate it into a single location.
Now the `DeployConfig` has the L1 contract addresses
required to build a L2 genesis. These addresses do not
need to be present in the JSON file itself and instead
can be read from disk dynamically.

Also add a check function that will error on a config
that is not sane. This can be made more strict in the
future, currently do not want to break anything in the
process of introducing this.

### CODE REVIEW

**Review by trianglesphere — APPROVED** [review]

**Review by protolambda — APPROVED** [review]
LGTM, updating the hardhat deploy configs with the new l1 recipient addresses would be nice to add though.

**Comment by tynes** [comment]
I've updated the deploy config files to contain the correct values

File: op-node/cmd/genesis/cmd.go [hunk 1]
```diff
@@ -52,6 +52,10 @@ var Subcommands = cli.Commands{
 				return err
 			}
 
+			if err := config.Check(); err != nil {
```
Comment (mslipper) [inline op-node/cmd/genesis/cmd.go:55]: Is this check duplicated on line 145 below?

**Review by mslipper — APPROVED** [review]

**Comment by mslipper** [comment]
Looks like the devnet has an expected failure given these changes:

```
CRIT [11-08|17:46:05.867] Application failed                       message="invalid deploy config: ProxyAdminOwner cannot be address(0)"
```

**Comment by tynes** [comment]
> Looks like the devnet has an expected failure given these changes:
> 
> ```
> CRIT [11-08|17:46:05.867] Application failed                       message="invalid deploy config: ProxyAdminOwner cannot be address(0)"
> ```

Will fix each issue

File: op-node/cmd/genesis/cmd.go (hunk 1 above)
Comment (tynes) [reply op-node/cmd/genesis/cmd.go:55]: That is a different command

**Comment by tynes** [comment]
Getting a segfault in `op-geth` in `TestL2Verifier_SequenceWindow`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xbd10bf]

goroutine 5087 [running]:
github.com/ethereum/go-ethereum/core/types.(*Block).NumberU64(...)
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v0.0.0-20221104231810-30db39cae2be/core/types/block.go:299
github.com/ethereum/go-ethereum/core/txpool.(*TxPool).reset(0xc006406600, 0xc01165ed80, 0xc01165fd40)
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v0.0.0-20221104231810-30db39cae2be/core/txpool/txpool.go:1268 +0xf3f
github.com/ethereum/go-ethereum/core/txpool.(*TxPool).runReorg(0xc006406600, 0xc0171e10e0, 0xc0026213e0, 0x0, 0xc00c74cf90)
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v0.0.0-20221104231810-30db39cae2be/core/txpool/txpool.go:1172 +0x26b
created by github.com/ethereum/go-ethereum/core/txpool.(*TxPool).scheduleReorgLoop
	/go/pkg/mod/github.com/ethereum-optimism/op-geth@v0.0.0-20221104231810-30db39cae2be/core/txpool/txpool.go:1101 +0x1b3
```

**Comment by trianglesphere** [comment]
> invalid deploy config: L2CrossDomainMessengerOwner cannot be address(0)

@tynes this is a known flake in the action tests

**Comment by tynes** [comment]
Flake with the devnet + deploy:

```
Starting withdrawal
An unexpected error occurred:

Error: transaction failed [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transactionHash="0x3b7453282ec8cb1f26f0cb8b98f5a2e441589d3716fd0850f4c52519371e6849", transaction={"type":2,"chainId":901,"nonce":1,"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x59685f30"},"gasPrice":null,"gasLimit":{"type":"BigNumber","hex":"0x028c73"},"to":"0x4200000000000000000000000000000000000010","value":{"type":"BigNumber","hex":"0x00"},"data":"0x32b7006d0000000000000000000000007c6b91d9be155a6db01f749217d76ff02a7227f20000000000000000000000000000000000000000000000000de0b6b3a7640000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000000","accessList":[],"hash":"0x3b7453282ec8cb1f26f0cb8b98f5a2e441589d3716fd0850f4c52519371e6849","v":0,"r":"0xd368ebcf1b0947e99b046acbddd5cd507ce3c1986fdc163f36bb171ca58290fb","s":"0x00d94dbc489a905ed61385087a4acb56267b36d1d2762a82688643789e7c9341","from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","confirmations":0}, receipt={"to":"0x4200000000000000000000000000000000000010","from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","contractAddress":null,"transactionIndex":2,"gasUsed":{"type":"BigNumber","hex":"0xa42d"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x368b7c99a293adc3319f48ee0306329b2d9cd45fa3454581b811a256838f3ed9","transactionHash":"0x3b7453282ec8cb1f26f0cb8b98f5a2e441589d3716fd0850f4c52519371e6849","logs":[],"blockNumber":76,"confirmations":22,"cumulativeGasUsed":{"type":"BigNumber","hex":"0x11c43d"},"effectiveGasPrice":{"type":"BigNumber","hex":"0x59685f30"},"status":0,"type":2,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.7.1)
    at Logger.makeError (/home/circleci/project/node_modules/@ethersproject/providers/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
```

This has become really flakey lately, not sure why

**Pull request merged**